### DISCUSSION TITLE
How to Decrease Number of Timesteps for RF Plasma?

### DISCUSSION THREAD

**Discussion by ghost** [shannon-lab/zapdos#113] [Q&A Plasma Transport] [answered]
I'm trying to build a simulation of an RF plasma jet, but I'm quickly realizing a major pitfall: there are WAY too many timesteps to cover anything close to 1 second (ideally longer) with a teeny-tiny timestep on the order of 1e-9 seconds. Is there any way to significantly increase the time-step while still approximating the RF behavior?

**Comment by csdechant** [comment, marked as answer]
The small time steps is one for the main limitations for RF plasma simulation in general. One needs the time resolution to capture the effects of the oscillating plasma sheath. We have some acceleration methods, but they are currently for metastable species. We also use HPCs to help reduce the wall time of the simulations. One question I have is what exactly are you trying to capture in this model? Currently one of the undergrads at NCSU, @gsgall, is helping with coupling Zapdos with MOOSE's Navier-Stokes module. When coupling with Navier-Stokes, sub-cycling can used, such that Zapdos resolve the plasma fluid equation for so many RF cycles every so many time steps of the Navier-Stokes solve.

**Reply by ghost** [reply]
We're trying to capture the steady-state downstream behavior of the jet. We're trying to formulate a simulation method both with and without the RF by making the approximation that far enough down-stream the potential changes from the RF are negligible, but this forces us to make assumptions about the plasma densities along boundaries which seems to conflict numerically with sheath boundaries, especially at intersections.

**Reply by csdechant** [reply]
If you are look at steady-state downstream behavior, then you might want to look into sub-cycling. In MOOSE user can have a main app run at relatively large time step and have a sub app run at desired times with smaller time steps (e.g. only solving for the electrons / mean energy in a sub app for X RF cycles to hit periodic steady-state and feeding their information to the main app, then resolving for the electron / mean energy when needed). The electrons / mean energy should hit a pseudo periodic steady-state within a few thousand RF cycles, or so, for every sub-app run.

**Reply by ghost** [reply]
So in this case, the main app would be the fluid model and the sub-app would be the RF circuit/electrode?

**Reply by csdechant** [reply]
I was thinking more of having both apps being the fluid model. The main app having just heavy species that can take the larger time step and have the needed electron data as some type of time averaging over an RF cycle. The sub app would solve all fluid equations at a dt=1e-9 until the rate of growth for the electrons is negligible. Just keep in mind, I am just thinking of ideas (for example, you might need to be careful when averaging the electron data for the main app, so to not create some type of artificial effects).
